Packet++/header/NflogLayer.h

```diff
@@ -70,6 +70,17 @@ namespace pcpp
  		NFULA_HWLEN					= 17,
 	};
 
+	static constexpr bool is_powerof2(int v) {
+		return v && ((v & (v - 1)) == 0);
+	}
+
+	// Only works for alignment with power of 2
+	static int align(int number, int alignment)
+	{
+		int mask = alignment - 1;
+		return (number + mask) & ~mask;
+	}
```

Comment:
Few comments:
1. We don't use `is_powerof2` anywhere... it's probably not really needed, we can check it inside the `align()` method
2. We should think what's the best place for `align()`. For now I think it can be a private method of `NflogTlv`. Or we can put it in `GeneralUtils.h` (under Common++). Let's see what @clementperon thinks

Reply:
1) It makes the code cleaner to read IMO (there is the static assert missing)
2) yes in Utils its a good idea

Reply:
It didn't compile with static assert saying it is not a const expression :( I'm open to suggestions if it can be fixed.

Reply:
@sashashura what do you think about:

```
diff --git a/Packet++/header/NflogLayer.h b/Packet++/header/NflogLayer.h
index ef8d991c..7d209260 100644
--- a/Packet++/header/NflogLayer.h
+++ b/Packet++/header/NflogLayer.h
@@ -75,8 +75,10 @@ namespace pcpp
        }
 
        // Only works for alignment with power of 2
-       static int align(int number, int alignment)
+       template <int alignment>
+       static const int align(int number)
        {
+               static_assert(is_powerof2(alignment), "Alignment must be a power of 2");
                int mask = alignment - 1;
                return (number + mask) & ~mask;
        }
@@ -115,7 +117,7 @@ namespace pcpp
                size_t getTotalSize() const
                {
                        // as in https://github.com/the-tcpdump-group/libpcap/blob/766b607d60d8038087b49fc4cf433dac3dcdb49c/pcap-util.c#L371-L374
-                       return align(m_Data->recordLen, 4);
+                       return align<4>(m_Data->recordLen);
                }
 
                /**
```